state_db.js

```diff
@@ -303,11 +304,37 @@ class StateDB {
     }
 
     async verify(root, queries, proof) {
-        return new Promise((resolve, reject) => {
-            state_db_verify.call(this._db, root, queries, proof, (err, result) => {
-                if (err) {
-                    return reject(err);
+        return new Promise((resolve) => {
+            state_db_verify.call(this._db, root, queries, proof, (_, result) => {
+                resolve(result);
+            });
+        });
+    }
+
+
+    async verifyInclusionProof(root, queries, proof) {
+        return new Promise((resolve) => {
+            for (let i = 0; i < queries.length; i++) {
+                if (!isInclusion(queries[i], proof.queries[i])) {
+                    resolve(false);
```

Comment:
i think you need return here. Also this for loop can be done out of promise

Reply:
Done.

Reply:
>Also this for loop can be done out of promise

Has this been already considered in the last commit?

Reply:
Yes, already done.